codegen/src/botocore.in.rs

```diff
@@ -85,6 +85,10 @@ impl Service {
         }
     }
 
+    pub fn shape_for_member<'a>(&'a self, member: &Member) -> Option<&'a Shape> {
```

Comment:
Added this function on the `service` type so when we're deep in somewhere and need to look up a shape for a child member we can do so.  Which causes us to need to pass the `service` object around a bit more.